pkg/florist/user_test.go

```diff
@@ -1,32 +1,107 @@
 package florist_test
 
 import (
+	"fmt"
+	"math/rand/v2"
+	"path"
+	"strconv"
+	"strings"
 	"testing"
 
 	"github.com/marco-m/florist/pkg/florist"
-	"github.com/marco-m/rosina/assert"
 )
 
-func TestUserAddSuccessVM(t *testing.T) {
+func TestUserAddSuccess(t *testing.T) {
 	florist.SkipIfNotDisposableHost(t)
 
-	err := florist.UserAdd("beppe")
+	// Ensure user doesn't exist.
+	name := "florist-" + strconv.Itoa(rand.IntN(100_000)+200_000)
 
-	assert.NoError(t, err, "florist.UserAdd")
+	err1 := florist.UserAdd(name, florist.UserAddOpts{})
+	if err1 != nil {
+		t.Errorf("\nadding non-existing user:\nhave error: %s\nwant: <no error>", err1)
+	}
+
+	// Now the user is already present, this should not be an error.
+	err2 := florist.UserAdd(name, florist.UserAddOpts{})
+	if err2 != nil {
+		t.Errorf("\nadding existing user:\nhave error: %s\nwant: <no error>", err1)
+	}
+
+	// The home directory has been created.
+	homeDir := path.Join("/home", name)
+	exists, err := florist.FileExists(homeDir)
+	if !exists {
+		t.Error("home directory does not exist:", homeDir)
+	}
+	if err != nil {
+		t.Errorf("\nwhile checking for existence of %s: %s", homeDir, err)
+	}
+}
+
+func TestUserAddSystemSuccess(t *testing.T) {
```

Comment:
Are you lacking 1 test? that test that configuring the UID to a value actually creates a user (and group) with that UID?